hw/bootrom/snitch/snitch_startup.c

```diff
@@ -43,19 +43,19 @@ void cluster_return(uint32_t ret) {
         *((volatile uint32_t *)(SOC_CTRL_BASE + CHIMERA_SNITCH_CLUSTER_0_RETURN_REG_OFFSET)) =
             retVal;
         break;
-    case 10:
+    case 1 + CLUSTER_0_NUMCORES:
         *((volatile uint32_t *)(SOC_CTRL_BASE + CHIMERA_SNITCH_CLUSTER_1_RETURN_REG_OFFSET)) =
             retVal;
         break;
-    case 19:
+    case 1 + CLUSTER_0_NUMCORES + CLUSTER_1_NUMCORES:
         *((volatile uint32_t *)(SOC_CTRL_BASE + CHIMERA_SNITCH_CLUSTER_2_RETURN_REG_OFFSET)) =
             retVal;
         break;
-    case 28:
+    case 1 + CLUSTER_0_NUMCORES + CLUSTER_1_NUMCORES + CLUSTER_2_NUMCORES:
         *((volatile uint32_t *)(SOC_CTRL_BASE + CHIMERA_SNITCH_CLUSTER_3_RETURN_REG_OFFSET)) =
             retVal;
         break;
-    case 37:
+    case 1 + CLUSTER_0_NUMCORES + CLUSTER_1_NUMCORES + CLUSTER_2_NUMCORES + CLUSTER_3_NUMCORES:
```

Comment:
yeah, I thought about this issue, too. The problem is that case patterns need to be compile-time constant, so any expression that computes this would need to be a macro operating on compile-time constant data. This gets messy quick, so I would leave it for a later update.

Reply:
Sounds good, just figured I'd mention it.